native_client/deepspeech.cc

```diff
@@ -273,7 +311,13 @@ StreamingState::finishStream()
     processBatch(batch_buffer, batch_buffer.size()/model->mfcc_feats_per_timestep);
   }
 
-  return model->decode(accumulated_logits);
+  if (extended_output) {
```

Comment:
How expensive is it to compute those informations? I'm not a big fan of changing the API just to expose this boolean, I'd rather we build the information always, as a datastructure and not as a JSON string, and maybe extend API (but not change current functions) to have a helper to produce a JSON string.

Reply:
This is already the case - JSON output is handled separately and you can do extra processing on the metadata if needed.

Reply:
@dabinat I'm not sure I understand you, the current reading I can make of this code does not reveal that those values are computed each time.

Reply:
Yeah, I think you’re right. The internal state should stay the same and the flag should only control output.

Reply:
@dabinat Even for the output I'm not sure. Store the `std::vector<META_DICT>` into `StreamingState`, and why not introduce a `StreamingState::finishStreamWithTiming` ? I'm not 100% sure on the way to extend that part. I'd like opinion from @reuben there.

Reply:
Or `StreamingState::finishStreamExtended` a there we can expose a more public datastructure. Or just a JSON string and it's much easier to deal with for compatibility.